src/js/mixins/SaveStateMixin.js

```diff
@@ -19,6 +19,10 @@ const SaveStateMixin = {
   },
 
   componentWillUnmount() {
+    this.saveState();
+  },
+
+  saveState() {
```

Comment:
we need to prefix all the methods / properties that come from mixins with the name. in this case, it would be `saveState_METHODNAME`. similar to how we do `saveState_key` above